src/main/java/liqp/nodes/BlockNode.java

```diff
@@ -79,6 +85,44 @@ private String asString(Object value, TemplateContext context) {
             ZonedDateTime time = asTemporal(value, context);
             return rubyDateTimeFormat.format(time);
         } else {
+            return getValueAsString(value, false);
+        }
+    }
+
+    private String mapToString(Map<Object, Object> map) {
```

Comment:
why we are replacing chars manually ? can we not convert this map to JSONObject ? and return string ?

Reply:
right Kartik.
I have tried that. converting to JSONObject to String again gives "=" in the string which we don't want.

Reply:
Please use org.json.simple JSONObject Library

Reply:
updated